app/assets/v2/js/grants/index.js

```diff
@@ -23,6 +23,36 @@ $(document).ready(() => {
 
 });
 
+function debounce(func, wait) {
+  let timeout;
+  let immediate;
+
+  return function() {
+    // reference the context and args for the setTimeout function
+    let context = this; let
+        args = arguments;
+    
+    // allow user-triggered debounce from the func's supplied argument
+    if (args[0].debounce !== undefined && args[0].debounce === true)
+      immediate = false;
+    else if (args[0].debounce === undefined || args[0].debounce === false)
+      immediate = true;
```

Comment:
Why are we applying debounce options as a param to the debounced func (this func shouldn't be aware that it is being debounced)?